server/src/http_server/pages/blog/md/html.rs

```diff
@@ -287,13 +291,25 @@ impl IntoHtml for Emphasis {
 }
 
 impl IntoHtml for Image {
-    fn into_html(
-        self,
-        _config: &AppConfig,
-        _context: &SyntaxHighlightingContext,
-    ) -> Result<Markup> {
+    fn into_html(self, config: &AppConfig, context: &SyntaxHighlightingContext) -> Result<Markup> {
+        let article_path = context
+            .current_article_path
+            .as_deref()
+            .unwrap_or("unknown")
+            .split('/')
+            .next()
+            .unwrap_or("unknown")
+            .replace(' ', "-"); // Replace spaces with hyphens
+
+        let image_url = format!(
+            "{}/posts/{}/{}",
+            config.base_url.trim_end_matches('/'),
+            article_path,
+            self.url.trim_start_matches("../")
```

Comment:
Would be nice to avoid this path parsing too!

Could we use some Rust Path structs/methods to like join the paths together? I feel like if we told it to join the paths and one has a `../` it will do the 'correct' thing and move up a directory there! 🤞